packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -1082,50 +1148,56 @@ class OverlayPortal extends RenderObjectWidget {
   RenderObject createRenderObject(BuildContext context) => _RenderLayoutSurrogateProxyBox();
 }
 
+@immutable
+class _OverlayChild {
```

Comment:
This class could probably use some documentation. Some questions to answer:

* Is `widget` the current configuration of `element`?
* Is `slot` the slot `element` has in its parent?
   *  Why is `slot` different from `element.slot`? Or is it the same?

Reply:
Actually, the purpose of this class is not entirely clear to me...

Reply:
The class is to make sure these things are non-null and the slot is properly typed.

Reply:
Removed since it only introduces 2 down casts